partner_delivery_schedule/models/partner_delivery_schedule.py

```diff
@@ -0,0 +1,74 @@
+# Copyright 2018 Tecnativa - Sergio Teruel
+# License AGPL-3.0 or later (https://www.gnu.org/licenses/agpl).
+from odoo import _, api, fields, models
+from odoo.exceptions import ValidationError
+
+
+class DeliverySchedule(models.Model):
+    _name = 'delivery.schedule'
+
+    name = fields.Char(required=True)
+    hour_from = fields.Float(string='From')
+    hour_to = fields.Float(string='To', default=24.00)
+    monday = fields.Boolean()
+    tuesday = fields.Boolean()
+    wednesday = fields.Boolean()
+    thursday = fields.Boolean()
+    friday = fields.Boolean()
+    saturday = fields.Boolean()
+    sunday = fields.Boolean()
+
+    @api.constrains('hour_from', 'hour_to')
+    def _check_hour_interval(self):
+        if (self.hour_from < 0.0 or self.hour_to > 24.0 or
+                self.hour_from >= self.hour_to):
+            raise ValidationError(_(
+                'Error ! You can not set hour_from greater or equal '
+                'than hour_to .'))
+        return True
+
+    @api.constrains(
+        'monday', 'tuesday', 'wednesday', 'wednesday', 'thursday', 'friday',
+        'saturday', 'sunday'
+    )
+    def _check_day_selected(self):
+        if not any([self[x[0]] for x in self._days_of_week()]):
+            raise ValidationError(
+                _('Error ! You must set one day to delivery.')
+            )
+        return True
+
+    def _days_of_week(self):
+        return [
+            ('monday', _('Monday')),
```

Comment:
You should add a translatable term for the initial of each day, as it's not enough taking the first letter of this. Example: in Spanish, we have `Martes` and `Miércoles`, which will display in both `M`

Reply:
Ok, but I have taken into account the two first letterf for every day.. so Martes is 'Ma' and Miercoles is 'Mi'

Reply:
I see... but it's better to use only one letter for not doing the m2m tag very long, although you have the option to be more if you want to translate it that way

Reply:
Ohh!!! sorry but Odoo do not generate a traslatable term  one word... (Of course), so if in english is M (monday) the translatable term not exists.., when a user check all days the name_get set to 'all days'..
I think that there is enought available space into m2m_tag field..

Reply:
![seleccion_020](https://user-images.githubusercontent.com/7689807/44979926-b69ff600-af6e-11e8-8b68-3280e3814637.jpg)

Reply:
OK, we let it this way